src/main/scala/beam/utils/mapsapi/googleapi/GoogleAdapter.scala

```diff
@@ -91,9 +91,11 @@ class GoogleAdapter(apiKey: String, outputResponseToFile: Option[Path] = None, a
   }
 
   private def parseRoutes(jsRoutes: Seq[JsValue]): Seq[Route] = {
-    jsRoutes.map { route =>
-      val firstAndUniqueLeg = (route \ "legs").as[JsArray].value.head
-      parseRoute(firstAndUniqueLeg.as[JsObject])
+    jsRoutes.flatMap { route =>
+      (route \ "legs").as[JsArray].value.headOption match {
```

Comment:
Wouldn't a map be better here?

Reply:
Agree. I did the change as you suggested